docs/dominio.md

```diff
@@ -0,0 +1,8 @@
+# HorseMatch: Dominio
+
+## Identificación de clases
+Atendiando a las historias de usuario redactadas se han identificado la clase **caballo** y la clase **jinete**, que conforman el dominio. La clase jinete será implementada inicialmente como inmutable, por lo que representará un _objeto-valor_, mientras que caballo podría variar algunos de sus datos miembros, por lo que es mutables y por tanto una _entidad_.
+
+**_¿Porque podría variar los datos miembros de la clase caballo?_**
+En la HU2 #4, se habla de que los caballos pueden estar descansando o no, por lo que en la clase caballo habría un dato miembro que represente la disponibilidad del caballo y que podría cambiar en función de si el caballo está descansado o no.
```

Comment:
Un "caballo" no va a estar descansado o no. El caballo será el que sea, con nombre y con dificultad (siempre que esta no varíe).
El *estado* del caballo es lo que hay que manejar desde una entidad, que de hecho tendrá que manejar el estado de todos los caballos. Hay varias formas de hacerlo, pero trabajar con objetos mutables siempre es problemático, porque hay que comprobar qué parte es inmutable, serializar/desserializar o cachear, y múltiples problemas más. Por eso en DDD hay que buscar los objetos valor (inmutables), que es lo que no estáis haciendo.

Reply:
Entonces, ¿el estado en el que está el caballo, no sería necesario implementarlo en esta fase?

Reply:
Si no lo haces, esto se queda en nada... 
Pero es cosa tuya decidir el alcance del PR.

Reply:
Osea podria ser conveniente tener una clase **CaballosDisponibles** que tenga un lista de caballos y donde para cada caballo guarde el estado del caballo.

Reply:
Podría ser conveniente que crearas un issue que plantee el problema, y propongas código que lo resuelva.